src/ImageSharp/ColorSpaces/Conversion/ColorSpaceConverter.Adapt.cs

```diff
@@ -0,0 +1,198 @@
+﻿// <copyright file="ColorSpaceConverter.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.ColorSpaces.Conversion
+{
+    using System;
+
+    using ImageSharp.ColorSpaces;
+    using ImageSharp.ColorSpaces.Conversion.Implementation.Rgb;
+
+    /// <content>
+    /// Performs chromatic adaptation on the various color spaces.
+    /// </content>
+    public partial class ColorSpaceConverter
```

Comment:
Yeah, a general bulk conversion could be really useful in some scenarios (like Jpeg), but I see very high chance that it's also possible to optimize the simple conversion. There are chains of calls on heap object instances. There is an `IColorConversion<in T, out TResult>` interface implemented by stateless objects, and used 99% oft the time through the concrete instances. I'm sure it's possible to find a better way to do this. 

Some concreate ideas:
- Making `IColorConversion.Convert()` implementations invokable through static methods. 
- Review of `IChromaticAdaptation`. Currently it introduces lot's of non-inlinable virtual method calls.
- Evalute possibilities to turn some of the classes into `struct`-s, and play with generic constraints to make more of the invocations inlineable